scripts/cleanup_submissions/cleanup_submissions.wdl

```diff
@@ -0,0 +1,245 @@
+  		call httpBatchDelete { input: fofn=fofn, 
+  		                           bucketName=getFilesOfSubmissions.bucketName, 
+  		                           svcActKeyJson=svcActKeyJson 
+  		}
+  	}
+  }
+
+  output {
+  	Array[File] filesToDelete = getFilesOfSubmissions.filesToDelete
+  }
+}
+
+task getSubmissionsInWorkspace {
+	File svcActKeyJson
+	String workspaceProject
+    String workspaceName
+
+    command {
+gcloud auth activate-service-account --key-file=${svcActKeyJson}
+export GOOGLE_APPLICATION_CREDENTIALS=${svcActKeyJson}
+
+python <<CODE
+# make use of the common library
+from common import *
+
+credentials = GoogleCredentials.get_application_default()
+print "Using Google client id:", credentials.client_id
+
+submissions_request = firecloud_api.list_submissions('${workspaceProject}', '${workspaceName}')
+
+if submissions_request.status_code != 200:
+	fail("Unable to list submissions for %s/%s  at  %s" (workspaceProject, workspaceName, get_fc_url()))
+
+submissions_json = submissions_request.json()
+terminal_submission_ids = map(lambda s: s["submissionId"], filter(lambda s: s["status"] in ["Aborted", "Done"], submissions_json))
+
+with open('submissions_to_clean.txt', 'w') as out_file:
+	out_file.write('\n'.join(terminal_submission_ids))
+CODE
+	}
+
+	output {
+		Array[String] submissionIds = read_lines("submissions_to_clean.txt")
+	}
+	
+	runtime {
+		docker: "broadinstitute/firecloud-tools"
+    	memory: 1
+    	disks: "local-disk 1 HDD"
+	}
+
+
+}
+
+task getFilesOfSubmissions {
+   File svcActKeyJson
+   String workspaceProject
+   String workspaceName
+   Array[String] submissionIds
+
+command {
+
+gcloud auth activate-service-account --key-file=${svcActKeyJson}
+export GOOGLE_APPLICATION_CREDENTIALS=${svcActKeyJson}
+    
+python <<CODE
+    # setup so we can import from the /scripts directory from the firecloud-tools repo so we can
+    # make use of the common library
+from common import *    
+workspaceProject = "${workspaceProject}"
+workspaceName = "${workspaceName}"
+
+
+credentials = GoogleCredentials.get_application_default()
+print "Using Google client id:", credentials.client_id
+
+workspace_request = firecloud_api.get_workspace(workspaceProject, workspaceName)
+ 
+if workspace_request.status_code != 200:
+	fail("Unable to find workspace: %s/%s  at  %s --- %s" % (workspaceProject, workspaceName, get_fc_url(), workspace_request.text))
+ 
+workspace = workspace_request.json()
+workspace_attributes = workspace["workspace"]["attributes"]
+
+bucketName = workspace["workspace"]["bucketName"]
+with open('bucket_name.txt', 'w') as out_file:
+	out_file.write(bucketName)
+
+entity_types_json = firecloud_api.list_entity_types(workspaceProject, workspaceName).json()
+entities_json = []
+for entity_type in entity_types_json:
+	entities_json.extend(firecloud_api.get_entities(workspaceProject, workspaceName, entity_type).json())
+
+
+submissionIds = ['${sep="','" submissionIds}']
```

Comment:
what exactly is going on here?

Reply:
Turning the WDL Array into a Python list